server/datastore/mysql/microsoft_mdm.go

```diff
@@ -172,7 +172,7 @@ func (ds *Datastore) mdmWindowsInsertCommandForHostsDB(ctx context.Context, tx s
 func (ds *Datastore) mdmWindowsInsertHostCommandDB(ctx context.Context, tx sqlx.ExecerContext, hostUUIDOrDeviceID, commandUUID string) error {
 	stmt := `
 INSERT INTO windows_mdm_command_queue (enrollment_id, command_uuid)
-VALUES ((SELECT id FROM mdm_windows_enrollments WHERE host_uuid = ? OR mdm_device_id = ?), ?)
+VALUES ((SELECT id FROM mdm_windows_enrollments WHERE host_uuid = ? OR mdm_device_id = ? ORDER BY created_at DESC LIMIT 1), ?)
```

Comment:
Note that this doesn't fix/prevent having duplicate host_uuid/mdm_device_id in `mdm_windows_enrollments`, current understanding is that this should not happen for real devices, is a VM quirk (see https://github.com/fleetdm/fleet/issues/20764#issuecomment-2343916289). This only fixes the sub-query so that the cron to deliver profiles will not fail if that unexpected situation arises.